src/Components/error-boundary/error-boundary.component.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react';
+
+import { ErrorImageOverlay, ErrorImageContainer, ErrorImageText } from './error-boundary.styles';
+
+export class ErrorBoundary extends Component {
+	constructor () {
+		super();
+
+		this.state = {
+			hasErrored: false,
+		};
+	}
+
+	static getDerivedStateFromError (error) {
+		return { hasErrored: true };
+	}
+
+	componentDidCatch (error, info) {
+		console.log(error);
```

Comment:
are we not supposed to set state here, the error to be true?

Reply:
No, that is the initial state(it has to be false otherwise it will always be error). The state lets us know if any of children in error boundary has actually thrown an error.

On the other hand,getDerivedStateFromError method will get the literal error that was thrown if any children inside of this error boundary does throw an error. Then we make sure to return from this method object representing the new state that we want to set locally